src/test/scala/is/hail/methods/LinearRegressionBurdenSuite.scala

```diff
@@ -0,0 +1,248 @@
+    */
+
+    // Values from R: fit <- lm(Pheno ~ Gene1 + Cov1 + Cov2, data=df)
+    val linregMapR: Map[Int, IndexedSeq[java.lang.Double]] = Map(
+      1 -> IndexedSeq(-0.08164, 0.15339, -0.532, 0.6478),
+      2 -> IndexedSeq(-0.09900, 0.17211, -0.575, 0.6233),
+      3 -> IndexedSeq(0.01558, 0.18323, 0.085, 0.940))
+
+    val sampleMapR: Map[Int, IndexedSeq[java.lang.Double]] = Map(
+      1 -> IndexedSeq(0.0, 5.0, 0.0, 4.0, 0.0, 1.0),
+      2 -> IndexedSeq(0.0, 4.0, 0.0, 4.0, 0.0, 0.0),
+      3 -> IndexedSeq(0.0, 5.0, 3.0, 7.0, 3.0, 1.0))
+
+    assertMap(linregMapR, linregMap, 1e-3)
+    assertMap(sampleMapR, sampleMap, 1e-6)
+  }
+
+  @Test def testWeightedSumWithImputation() {
+
+    val (linregKT, sampleKT) = vdsBurden
+      .filterSamplesExpr("isDefined(sa.pheno.Pheno) && isDefined(sa.cov.Cov1) && isDefined(sa.cov.Cov2)")
+      .variantQC()
+      .linregBurden("gene", "va.genes",
+      "gs.map(g => va.weight * orElse(g.gt.toDouble, 2 * va.qc.AF)).sum()",
+      "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1", "sa.cov.Cov2"))
+
+    val linregMap = makeMap(linregKT)
+    val sampleMap = makeMap(sampleKT)
+
+    /*
+    Variant A       B       C       D       E       F       Weight
+    1       0.0     1.0     0.0     0.0     0.0     1.0     1.0
+    2       1.0     2.0     1.0     2.0     0.0     0.0     2.0
+    3       0.0     .75     1.0     1.0     1.0     .75     3.0
+
+    Sample  Cov1    Cov2    Pheno   Gene1   Gene2   Gene3
+    A       0.0     -1.0    1.0     2.0     2.0     2.0
+    B       2.0     3.0     1.0     5.0     4.0     7.25
+    C       1.0     5.0     2.0     2.0     2.0     5.0
+    D       -2.0    0.0     2.0     4.0     4.0     7.0
+    E       -2.0    -4.0    2.0     0.0     0.0     3.0
+    F       4.0     3.0     2.0     1.0     0.0     3.25
+    */
+
+    // Values from R: fit <- lm(Pheno ~ Gene1 + Cov1 + Cov2, data=df)
+    val linregMapR: Map[Int, IndexedSeq[java.lang.Double]] = Map(
+      1 -> IndexedSeq(-0.2177, 0.1621, -1.343, 0.3115),
+      2 -> IndexedSeq(-0.2709, 0.1675, -1.617, 0.2473),
+      3 -> IndexedSeq(-0.05710, 0.21078, -0.271, 0.812))
+
+    val sampleMapR: Map[Int, IndexedSeq[java.lang.Double]] = Map(
+      1 -> IndexedSeq(2.0, 5.0, 2.0, 4.0, 0.0, 1.0),
+      2 -> IndexedSeq(2.0, 4.0, 2.0, 4.0, 0.0, 0.0),
+      3 -> IndexedSeq(2.0, 7.25, 5.0, 7.0, 3.0, 3.25))
+
+
+    assertMap(linregMapR, linregMap, 1e-3)
+    assertMap(sampleMapR, sampleMap, 1e-6)
+    }
+
+  @Test def testMax() {
+    val (linregKT, sampleKT) = vdsBurden.linregBurden("gene", "va.genes",
+      "gs.map(g => g.gt.toDouble).max()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1", "sa.cov.Cov2"))
+
+    val linregMap = makeMap(linregKT)
+    val sampleMap = makeMap(sampleKT)
+
+    /*
+    Variant A       B       C       D       E       F
+    1       0.0     1.0     0.0     0.0     0.0     1.0
+    2         .     2.0       .     2.0     0.0     0.0
+    3       0.0       .     1.0     1.0     1.0       .
```

Comment:
The method does no automatic imputation except inside linear regression. So there is no mean imputation in any of these tables, only that what the aggExpr explicitly does with orElse. But that's not my method, that's the user's choice.